smurfs/src/actions/actions.js

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios';
+//action types
+export const FETCH_SMURFS_START = 'FETCH_SMURFS_START';
+export const FETCH_SMURFS_SUCCESS = 'FETCH_SMURFS_SUCCESS';
+export const FETCH_SMURFS_FAILURE = 'FETCH_SMURFS_FAILURE';
+export const ADD_SMURFS = 'ADD_SMURFS';
+export const REMOVE_SMURFS = 'REMOVE_SMURFS';
+
+//action creator
+
+export const getSmurf = () => dispatch => {
+    dispatch({ type: FETCH_SMURFS_START });
+    axios.get('http://localhost:3333/smurfs')
+        .then(res => {
+            dispatch({ type: FETCH_SMURFS_SUCCESS, payload: res.data });
+        })
+        .catch(err => {
+            dispatch({ type: FETCH_SMURFS_FAILURE, payload: err.response })
+        })
+};
+
+export const addSmurf = smurf => dispatch => {
+    dispatch({ type: ADD_SMURFS });
```

Comment:
So for here, you dont actually need to call the dispatch. You are already calling the add smurfs down below in the axios call so this part here is redundant.